src/python/grpcio/grpc/_cython/_cygrpc/aio/iomgr/io_loop.pyx.pxi

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+import threading
+
+_SECONDS_PER_MILLISECOND = 0.001
+cdef _IOLoop _io_loop = None
+
+cdef class _IOLoop:
+
+    def __init__(self):
+        global _io_loop
+
+        # Only one instantiation is exepected        
+        assert _io_loop is None
+
+        self._asyncio_loop = None
+        self._io_ev = threading.Event()
+        self._loop_started_ev = threading.Event()
+
+        self._thread = threading.Thread(target=self._run_forever, daemon=True)
+        self._thread.start()
+
+        # Some attributes are initialitzated when
+        # the thread is really started, we wait till
+        # the whole Asyncio loop is initialized and
+        # ready to be used.
+        self._loop_started_ev.wait()
+
+        _io_loop = self
+
+    def _loop_started_cb(self):
+        self._loop_started_ev.set()
+
+    def _run_forever(self):
+        self._asyncio_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self._asyncio_loop)
+        self._asyncio_loop.call_soon(self._loop_started_cb)
+        try:
+            self._asyncio_loop.run_forever()
+        except Exception as exp:
+            sys.stderr("An error ocurred with the IO loop {}{}".format(exp, os.linesep))
```

Comment:
`sys.stderr.write`?

Reply:
I missed this one. Should we use `_LOGGER.error` or `_LOGGER.exception`?

Reply:
facepalm!

Ive used the `sys.stderr` because for an unknown reason `print` within Cyhton does not have support for the `file` kw, no idea the reason why.

I can use the `_LOGGER.exception` as it's suggested by Lidi. do you agree?

Reply:
Yeah. `_LOGGER` is probably better.

Reply:
+1 to using ``_LOGGER``